lib/active_shipping/carriers/ups.rb

```diff
@@ -450,15 +450,23 @@ def build_shipment_request(origin, destination, packages, options={})
             end
           end
 
-          # I don't know all of the options that UPS supports for labels
-          # so I'm going with something very simple for now.
+          # defaults to GIF
+          image_format = options[:image_format].nil? ? 'GIF' : options[:image_format].upcase
+
           xml.LabelSpecification do
+            # Support ZPL Files
+            if image_format == 'ZPL'
+              xml.LabelStockSize do
+                xml.Height('4')
+                xml.Width('6')
+              end
+            end
             xml.LabelPrintMethod do
-              xml.Code('GIF')
+               xml.Code(image_format)
             end
             xml.HTTPUserAgent('Mozilla/4.5') # hmmm
```

Comment:
Sorry I don't mean to overwhelm you with comments but I actually just noticed that the UPS docs also say that 'LabelImageFormat' is only required when the LabelPrintMethod == 'GIF' so it might make sense to include that in the conditional so it was something like
```
if image_format == 'GIF'
  xml.HTTPUserAgent('Mozilla/4.5')
  xml.LabelImageFormat(image_format) do
  xml.Code(image_format)
else
  xml.LabelStockSize do
    xml.Height('4')
    xml.Width('6')
  end
end
```